src/folderitemdelegate.cpp

```diff
@@ -178,12 +178,24 @@ void FolderItemDelegate::drawText(QPainter* painter, QStyleOptionViewItemV4& opt
 
   QPalette::ColorGroup cg = opt.state & QStyle::State_Enabled ? QPalette::Normal : QPalette::Disabled;
   if(opt.state & QStyle::State_Selected) {
-    painter->fillRect(selRect, opt.palette.highlight());
+    if(!opt.widget)
+      painter->fillRect(selRect, opt.palette.highlight());
     painter->setPen(opt.palette.color(cg, QPalette::HighlightedText));
   }
   else
     painter->setPen(opt.palette.color(cg, QPalette::Text));
 
+  if (opt.state & QStyle::State_Selected || opt.state & QStyle::State_MouseOver) {
+    if (const QWidget* widget = opt.widget) { // let the style engine do it
```

Comment:
I'm just concerned that the second condition will always be true due to the assignment
```const QWidget* widget = opt.widget```